### PR TITLE
`copy` for `BigInt`, `BigFloat` and `Rational`

### LINKED ISSUE #163: Why is there no mutable assignment operation?
I'd like it if I could do this:

```
mut = set!!(mut, num)
```

instead of this:

```
mut = zero!!(mut)
mut = add!!(mut, num)
```

Am I missing something?

### PR DESCRIPTION
**Pull request by nsajko**
Fixes #163.

### CODE REVIEW

**Comment by nsajko** [comment]
The failing Aqua lint is unrelated (Aqua is complaining about missing compat bounds for dependencies).

**Comment by odow** [comment]
> The failing Aqua lint is unrelated (Aqua is complaining about missing compat bounds for dependencies).

I can fix this

**Review by odow — COMMENTED** [review]
Looks okay to me, but this needs @blegat's review.

File: src/implementations/BigFloat.jl [hunk 1]
```diff
@@ -371,17 +389,7 @@ function buffered_operate_to!(
...
-            (:mpfr_set, :libmpfr),
-            Int32,
-            (Ref{BigFloat}, Ref{BigFloat}, Base.MPFR.MPFRRoundingMode),
-            out,
-            in,
-            Base.MPFR.ROUNDING_MODE[],
-        )
-        return nothing
-    end
+    set! = (o, i) -> operate_to!(o, copy, i)
```
Comment (blegat) [inline src/implementations/BigFloat.jl:392]: I think you need to keep the `local ` to avoid the performance issue of closures

Comment (nsajko) [reply src/implementations/BigFloat.jl:392]: Besides, the tests would have caught any extra allocations.

**Review by blegat — APPROVED** [review]
Thanks!

**Pull request merged**